test/03-updates.test.js

```diff
@@ -72,28 +63,22 @@ test('Pear.updates() should be called when restaging and releasing', async funct
   ok(seed2Key, `reseeded platform key (${seed2Key})`)
   ok(seed2Announced, 'reseed announced')
 
-  const awaitUpdates = async function (length, type = 'app') {
-    while (global.__PEAR_TEST__.updates[type]?.length < length) {
-      await new Promise(resolve => setTimeout(resolve, 100))
-    }
-
-    return global.__PEAR_TEST__.updates[type]
-  }.toString()
-  const update1 = await inspector.evaluate(`(${awaitUpdates})(1)`, { awaitPromise: true })
-  const update1Version = update1?.value?.[0]?.version
-  is(update1?.value?.length, 1, 'app updated after stage')
+  const update1 = await inspector.awaitPromise(update1Promise.objectId)
+  const update1Version = update1?.value?.version
   is(z32.encode(Buffer.from(update1Version?.key, 'hex')), key, 'app updated with matching key')
+  is(update1Version?.fork, 0, 'app version.fork is 0')
+  ok(update1Version?.length > 0, `app version.length is non-zero (v${update1Version?.fork}.${update1Version?.length})`)
 
   comment('releasing')
+  const update2Promise = await inspector.evaluate(awaitUpdate, { returnByValue: false })
   await helper.pick(helper.release(releaseOpts(testId, key), { close: false }), { tag: 'released' })
 
   comment('waiting for update')
-  const update2 = await inspector.evaluate(`(${awaitUpdates})(2)`, { awaitPromise: true })
-  is(update2?.value?.length, 2, 'app reupdated after release')
-
-  const update2Version = update2?.value?.[1]?.version
-  is(z32.encode(Buffer.from(update2Version?.key, 'hex')), key, 'app update2 with matching key')
-  ok(update2Version?.length > update1Version?.length, 'app version incremented')
+  const update2 = await inspector.awaitPromise(update2Promise.objectId)
+  const update2Version = update2?.value?.version
+  is(z32.encode(Buffer.from(update2Version?.key, 'hex')), key, 'app updated with matching key')
+  is(update2Version?.fork, 0, 'app version.fork is 0')
+  ok(update2Version?.length > update1Version?.length, `app version.length incremented (v${update2Version?.fork}.${update2Version?.length})`)
 
   await inspector.evaluate('global.__PEAR_TEST__.inspector.disable()')
```

Comment:
put this in inspector.close() (wrapped with a try/catch ignore)

Reply:
This line should now be in inspector._close